// @dfns/sdk/dfnsDelegatedApiClient.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DfnsDelegatedApiClient = void 0;
const auth_1 = require("./generated/auth");
const exchanges_1 = require("./generated/exchanges");
const feeSponsors_1 = require("./generated/feeSponsors");
const keys_1 = require("./generated/keys");
const networks_1 = require("./generated/networks");
const permissions_1 = require("./generated/permissions");
const policies_1 = require("./generated/policies");
const signers_1 = require("./generated/signers");
const wallets_1 = require("./generated/wallets");
const webhooks_1 = require("./generated/webhooks");
class DfnsDelegatedApiClient {
    constructor(apiOptions) {
        this.apiOptions = apiOptions;
    }
    get auth() {
        return new auth_1.DelegatedAuthClient(this.apiOptions);
    }
    get exchanges() {
        return new exchanges_1.DelegatedExchangesClient(this.apiOptions);
    }
    get feeSponsors() {
        return new feeSponsors_1.DelegatedFeeSponsorsClient(this.apiOptions);
    }
    get keys() {
        return new keys_1.DelegatedKeysClient(this.apiOptions);
    }
    get networks() {
        return new networks_1.DelegatedNetworksClient(this.apiOptions);
    }
    get permissions() {
        return new permissions_1.DelegatedPermissionsClient(this.apiOptions);
    }
    get policies() {
        return new policies_1.DelegatedPoliciesClient(this.apiOptions);
    }
    get wallets() {
        return new wallets_1.DelegatedWalletsClient(this.apiOptions);
    }
    get webhooks() {
        return new webhooks_1.DelegatedWebhooksClient(this.apiOptions);
    }
    get signers() {
        return new signers_1.DelegatedSignersClient(this.apiOptions);
    }
}
exports.DfnsDelegatedApiClient = DfnsDelegatedApiClient;
