// @dfns/sdk/dfnsApiClient.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DfnsApiClient = void 0;
const auth_1 = require("./generated/auth");
const exchanges_1 = require("./generated/exchanges");
const feeSponsors_1 = require("./generated/feeSponsors");
const keys_1 = require("./generated/keys");
const networks_1 = require("./generated/networks");
const permissions_1 = require("./generated/permissions");
const policies_1 = require("./generated/policies");
const signers_1 = require("./generated/signers");
const staking_1 = require("./generated/staking");
const wallets_1 = require("./generated/wallets");
const webhooks_1 = require("./generated/webhooks");
class DfnsApiClient {
    constructor(apiOptions) {
        this.apiOptions = apiOptions;
    }
    get auth() {
        return new auth_1.AuthClient(this.apiOptions);
    }
    get exchanges() {
        return new exchanges_1.ExchangesClient(this.apiOptions);
    }
    get feeSponsors() {
        return new feeSponsors_1.FeeSponsorsClient(this.apiOptions);
    }
    get keys() {
        return new keys_1.KeysClient(this.apiOptions);
    }
    get networks() {
        return new networks_1.NetworksClient(this.apiOptions);
    }
    get permissions() {
        return new permissions_1.PermissionsClient(this.apiOptions);
    }
    get policies() {
        return new policies_1.PoliciesClient(this.apiOptions);
    }
    get staking() {
        return new staking_1.StakingClient(this.apiOptions);
    }
    get signers() {
        return new signers_1.SignersClient(this.apiOptions);
    }
    get wallets() {
        return new wallets_1.WalletsClient(this.apiOptions);
    }
    get webhooks() {
        return new webhooks_1.WebhooksClient(this.apiOptions);
    }
}
exports.DfnsApiClient = DfnsApiClient;
